sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/encoders/RowEncoder.scala

```diff
@@ -288,7 +287,7 @@ object RowEncoder {
 
       StaticInvoke(
         ArrayBasedMapData.getClass,
-        ObjectType(classOf[Map[_, _]]),
+        ObjectType(classOf[scala.collection.immutable.Map[_, _]]),
```

Comment:
I don't know this bit well, but the comments on the JIRA indicate that immutable maps aren't supported. However, this change should be a no-op because Predef.Map is already immutable.Map right -- how does this address it?